backend/src/main/kotlin/outcobra/server/util/Extensions.kt

```diff
@@ -33,8 +33,14 @@ fun LocalDate.isAfterOrEqual(b: LocalDate): Boolean = this.isAfter(b) || this.is
  * @author Florian Bürgi
  * @since <since>
  */
-infix fun SchoolYear.doesNotOverlap(schoolYear: SchoolYear): Boolean =
-        this.id == schoolYear.id || !DateUtil.isOverlap(this.validFrom, this.validTo, schoolYear.validFrom, schoolYear.validTo)
+infix fun SchoolYear.doesNotOverlap(schoolYear: SchoolYear): Boolean {
+    if (this.id != schoolYear.id) {
+        val thisRange = DateRange(this.validFrom, this.validTo)
+        val otherRange = DateRange(schoolYear.validFrom, schoolYear.validTo)
+        return !(thisRange checkOverlap otherRange)
```

Comment:
I could do `'return !thisRange.checkOverlap(otherRange)` would you like that better?